### PR TITLE
chore: update NOTICE

### PR DESCRIPTION
**Pull request by kenji-miyake**
## Description

I updated `NOTICE` for AutoCore and Leo Drive since it seems they started using the format of `The Autoware Contributors`. (For example in https://github.com/autowarefoundation/autoware.universe/pull/1137.)

Related: https://github.com/autowarefoundation/autoware.ai/issues/2425

The format is from https://github.com/apache/solr/blob/0ba432371064586a900ae1e66b483f0685d4ef7f/NOTICE.txt.

@Sharrrrk @xmfcx Could you check if your license notations are appropriate? I've checked some of your files, but I'm not confident.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: NOTICE [hunk 1]
```diff
@@ -6,3 +6,9 @@ The Autoware Foundation (https://www.autoware.org/).
 
 This product includes code developed by TIER IV.
 Copyright 2017 TIER IV, Inc.
+
+This product includes code developed by AutoCore.
+Copyright 2022 AutoCore
+
+This product includes code developed by Leo Drive.
+Copyright 2022 Leo Drive Teknoloji A.Ş., Inc.
```
Comment (xmfcx) [inline NOTICE:14]:
```suggestion
Copyright 2022 Leo Drive Teknoloji A.Ş.
```
This is enough like this.

**Review by xmfcx — CHANGES_REQUESTED** [review]

**Comment by xmfcx** [comment]
@Sharrrrk could you also confirm the Autocore part please?

**Review by xmfcx — APPROVED** [review]

**Comment by kenji-miyake** [comment]
@Sharrrrk Friendly ping. :bow:

**Comment by kenji-miyake** [comment]
@Sharrrrk Could you review this, please?

File: NOTICE [hunk 2]
```diff
@@ -6,3 +6,9 @@ The Autoware Foundation (https://www.autoware.org/).
 
 This product includes code developed by TIER IV.
 Copyright 2017 TIER IV, Inc.
+
+This product includes code developed by AutoCore.
+Copyright 2022 AutoCore
```
Comment (Sharrrrk) [inline NOTICE:11]:
```diff
- Copyright 2022 AutoCore
+ Copyright 2022 AutoCore Technology (Nanjing) Co., Ltd.
```

**Review by Sharrrrk — CHANGES_REQUESTED** [review]

File: NOTICE (hunk 2 above)
Comment (kenji-miyake) [reply NOTICE:11]:
```suggestion
Copyright 2022 AutoCore Technology (Nanjing) Co., Ltd.
```

Comment (kenji-miyake) [reply NOTICE]: Updated in 65515f368493a316f8c32a11e5d39939226ec08c.

**Review by Sharrrrk — APPROVED** [review]
All good.

**Pull request merged**